skglm/datafits/single_task.py

```diff
@@ -126,6 +126,15 @@ def get_spec(self):
     def params_to_dict(self):
         return dict()
 
+    def raw_grad(self, y, Xw):
+        """Compute gradient of datafit w.r.t ``Xw``."""
+        return -y / (1 + np.exp(y * Xw)) / len(y)
+
+    def raw_hessian(self, y, Xw):
+        """Compute hessian of datafit w.r.t ``Xw``."""
```

Comment:
cosmit: Hessian, and it's the diagonal of the Hessian

Reply:
The Hessian is diagonal as the datafit is assumed to be separable, in particular for the logreg:

$$F(u) = \sum_{i=1}^n log(1+e^{-y_i u_i})$$

Reply:
yes I agree, but the Hessian is still a n by n matrix, while here you return only its diagonal entries 
anyways, not very important. I meant that you may want to clarify that you return only the diagonal because the Hessian is diagonal.